.github/workflows/inspektor-gadget.yml

```diff
@@ -2,7 +2,9 @@ name: Inspektor Gadget CI
 env:
   REGISTRY: ghcr.io
   CONTAINER_REPO: ${{ github.repository }}
-  GO_VERSION: 1.21.3
+  GO_VERSION: 1.22.0
+  # controller-gen with go >1.21 panics, but we can't update controller-gen itself
+  GO_VERSION_DOC_CHECK: 1.21.3
```

Comment:
The newer version of `controller-gen` without problems. But we need the old flags of the previous versions

Reply:
I did not investigate it too deep, but we might not be able to use other flags: https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.7.0
> Bump to kubernetes 1.22.1, remove support for legacy v1beta1 CRDs and webhooks

We use `trivialVersions` (support for v1beta1 CRDs) and it was removed in the linked PR of that release. But it might be a total other problem